src/main/java/utils/DriverManager.java

```diff
@@ -3,56 +3,127 @@
 import configurations.ConfigReader;
 import io.appium.java_client.AppiumDriver;
 import io.appium.java_client.android.AndroidDriver;
+import io.appium.java_client.ios.IOSDriver;
+import io.appium.java_client.ios.options.XCUITestOptions;
 import org.openqa.selenium.remote.DesiredCapabilities;
 
 import java.net.URI;
 import java.time.Duration;
 
 public class DriverManager {
-    private static AppiumDriver driver;
+    private static final ThreadLocal<AppiumDriver> driverThreadLocal = new ThreadLocal<>();
+    private static final Duration IMPLICIT_WAIT_TIMEOUT = Duration.ofSeconds(20);
 
-    private static final String DEVICE_NAME = ConfigReader.emulatorConfig.deviceName();
-    private static final String PLATFORM_NAME = ConfigReader.emulatorConfig.platformName();
-    private static final String APP_PACKAGE = ConfigReader.emulatorConfig.appPackage();
-    private static final String APP_ACTIVITY = ConfigReader.emulatorConfig.appActivity();
-    private static final String URL = ConfigReader.emulatorConfig.remoteURL();
-    private static final String AUTOMATION_NAME = ConfigReader.emulatorConfig.automationName();
-
-    private static DriverManager INSTANCE;
-
-    public static DriverManager getInstance() {
-        if (INSTANCE == null) {
-            INSTANCE = new DriverManager();
+    public static AppiumDriver getDriver() {
+        AppiumDriver driver = driverThreadLocal.get();
+        if (driver == null) {
+            driver = initDriver();
+            driverThreadLocal.set(driver);
         }
-        return INSTANCE;
+        return driver;
     }
 
-    public AppiumDriver initDriver() throws Exception {
+    private static AppiumDriver initDriver() {
```

Comment:
i would recommend to pass driver info as an argument  to this method, it should be clear and transparent what driver you are creating while with current implementation - it's not.